google/cloud/internal/rest_opentelemetry_test.cc

```diff
@@ -76,6 +80,28 @@ TEST(RestOpentelemetry, MakeSpanHttp) {
                                          secret.substr(0, 32))))));
 }
 
+TEST(RestOpentelemetry, InjectTraceContext) {
+  auto span_catcher = InstallSpanCatcher();
+  auto propagator =
+      std::make_shared<opentelemetry::trace::propagation::HttpTraceContext>();
+  // Set the global propagator.
+  opentelemetry::context::propagation::GlobalTextMapPropagator::
+      SetGlobalPropagator(
+          opentelemetry::nostd::shared_ptr<
+              opentelemetry::context::propagation::TextMapPropagator>(
+              propagator));
```

Comment:
FYI: I tested the gRPC version using a mock propagator.

https://github.com/googleapis/google-cloud-cpp/blob/be4216a5cb575ca303e8ea854a4a5b376b4048b0/google/cloud/internal/grpc_opentelemetry_test.cc#L91-L104

... but maybe that does not cover the line:

```cc
  auto current = opentelemetry::context::RuntimeContext::GetCurrent();
```

Reply:
I would have done the same with gRPC, since it is hard to get data out of `grpc::ClientContext`, but `RestContext` is more test friendly.